### PR TITLE
Test issues

### LINKED ISSUE #211: Command-line -t requires from wrong directory
In a directory other than `<PEGGY_ROOT>/bin`, this fails:

```sh
touch ast.js
echo '{{ require("./ast.js") }}\nfoo="1"' | peggy -t 1
```

the `eval` at the heart of the test routine is relative to the CLI script. I think there's a similar issue when we have `{output: "parser"}` in the API.

Some combination of `vm` and `require("module").createRequire` (when it's available) might be needed.  Probably also making `__dirname` and `__filename` match the outputFile if it's not "-", and using cwd and a synthetic name when it is "-".

In node 10, `createRequire` doesn't exist, but the undocumented `createRequireFromPath` does, as well as `new Module(path).require`.

I don't know how to fix this for `import`, but we can probably at least hack in `import.meta.url` same as `__filename`.

### PR DESCRIPTION
**Pull request by hildjj**
Fixes #211, #212, #213, #214.

### CODE REVIEW

**Comment by hildjj** [comment]
After several days to get fresh eyes, I re-looked at these commits, and they all seem adequate.  Any suggestions?

File: bin/peggy-cli.mjs [hunk 1]
```diff
@@ -285,10 +293,10 @@ export class PeggyCLI extends Command {
           }
         }
 
-        if (this.progOptions.test && this.progOptions.testFile) {
-          this.error("The -t/--test and -T/--test-file options are mutually exclusive.");
-        }
-        if (this.progOptions.test) {
+        if (typeof this.progOptions.test === "string") {
```
Comment (Mingun) [inline bin/peggy-cli.mjs:296]: I suppose that this check should fix #213? It is better to add a comment why exactly that check is required, or, in other words, why the `if (this.progOptions.test ...)` does not work

File: bin/peggy-cli.mjs [hunk 2]
```diff
@@ -404,7 +412,7 @@ export class PeggyCLI extends Command {
 
     if (this.outputFile === "-") {
       return Promise.resolve(
-        (!this.testFile && !this.testText) ? this.std.out : null
+        (!this.testFile && (typeof this.testText !== "string")) ? this.std.out : null
```
Comment (Mingun) [inline bin/peggy-cli.mjs:415]: The same question about `typeof` check -- it is better to explain it explicitly, why just `!this.testText` is not enough

File: bin/peggy-cli.mjs [hunk 3]
```diff
@@ -489,7 +497,49 @@ export class PeggyCLI extends Command {
...
     if (typeof this.testText === "string") {
       this.verbose("TEST TEXT:", this.testText);
-      const exec = eval(source);
+
+      const filename = this.outputJS
+        ? path.resolve(this.outputJS)
+        : path.join(process.cwd(), "stdout.js"); // Synthetic
+
+      const dirname = path.dirname(filename);
+      const m = new Module(filename, module);
```
Comment (Mingun) [inline bin/peggy-cli.mjs:506]: I suppose this is fix for #211. It is better to add an explanation of why and how it works (a link to the receipt if you cannot describe that by your words)

**Review by Mingun — APPROVED** [review]
The code looks good although I didn't dig into details of `vm`. Would be fantastic if you add comments in the highlighted places.

File: bin/peggy-cli.mjs (hunk 1 above)
Comment (hildjj) [reply bin/peggy-cli.mjs:297]: Made it clear that this is to catch `-t ""`.

File: bin/peggy-cli.mjs (hunk 3 above)
Comment (hildjj) [reply bin/peggy-cli.mjs:511]: Added a comment.  Let me know if you want more.

File: bin/peggy-cli.mjs (hunk 1 above)
Comment (Mingun) [reply bin/peggy-cli.mjs:297]: Nice

File: bin/peggy-cli.mjs (hunk 3 above)
Comment (Mingun) [reply bin/peggy-cli.mjs:511]: That's great!

**Review by Mingun — APPROVED** [review]

**Comment by hildjj** [comment]
Fixes #211.
Fixes #212.
Fixes #213.
Fixes #214.

**Pull request merged**